packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/amplify-meta-updaters.ts

```diff
@@ -100,3 +104,66 @@ export const getPostUpdateAuthMetaUpdater = (context: any) => async (resourceNam
   }
   return resourceName;
 };
+
+function getFrontEndConfig(authParameters: any) {
+  const frontEndConfig: any = {};
+
+  let loginMechanism: string[] = [];
+
+  if (authParameters.aliasAttributes) {
+    let aliasAttributes = authParameters.aliasAttributes.map((att: string) => att.toUpperCase());
+    loginMechanism = [...aliasAttributes];
+  }
+
+  if (authParameters.authProviders) {
+    authParameters.authProviders.forEach((provider: string) => {
+      let name = authProviderList.find(it => it.value === provider)?.name;
+
+      if (name) {
+        loginMechanism.push(name.toUpperCase());
+      }
+    });
+  }
+
+  if (loginMechanism.length) {
+    frontEndConfig.loginMechanism = loginMechanism;
+  }
+
+  if (authParameters.requiredAttributes) {
+    let requiredAttributes = authParameters.requiredAttributes.map((att: string) => att.toUpperCase());
+    frontEndConfig.signupAttributes = [...requiredAttributes];
+  }
```

Comment:
Assuming there's no functional difference between not setting the prop and setting it to undefined, all of these if statements can be replaced with something like:
```suggestion
frontEndConfig.signupAttributes = authParameters?.requiredAttributes ? authParameters.requiredAttributes.map((att: string) => att.toUpperCase()) : undefined
```